be/src/storage/persistent_index.cpp

```diff
@@ -3434,6 +3434,12 @@ bool PersistentIndex::_need_rebuild_index(const PersistentIndexMetaPB& index_met
         // Make sure l2 version equal to l2 version merged flag
         return true;
     }
+    if (index_meta.cpu_arch_id() != CPU_ARCH_ID && config::pindex_cpu_arch_migration_check) {
+        // If we found the migration between different CPU arch, we need to rebuild index.
```

Comment:
in this case, shall we delete the existing index file first to stop it from being used for updating/generate new tablet files?

Reply:
No, it you set `pindex_cpu_arch_migration_check` to true, index will be rebuilt after migration.